lychee-lib/src/utils/fragment_checker.rs

```diff
@@ -39,37 +43,28 @@ impl FragmentChecker {
     ///
     /// In all other cases, returns true.
     pub(crate) async fn check(&self, path: &Path, url: &Url) -> Result<bool> {
-        match (FileType::from(path), url.fragment()) {
-            (FileType::Markdown, Some(fragment)) => {
-                let url_without_frag = Self::remove_fragment(url.clone());
-                self.populate_cache_if_vacant(url_without_frag, path, fragment)
-                    .await
-            }
-            _ => Ok(true),
-        }
-    }
-
-    fn remove_fragment(mut url: Url) -> String {
-        url.set_fragment(None);
-        url.into()
-    }
+        let Some(fragment) = url.fragment() else {
+            return Ok(true)
+        };
+        let url_without_frag = Self::remove_fragment(url.clone());
 
-    /// Populates the fragment cache with the given URL if it
-    /// is not already in the cache.
-    async fn populate_cache_if_vacant(
-        &self,
-        url_without_frag: String,
-        path: &Path,
-        fragment: &str,
-    ) -> Result<bool> {
-        let mut fragment_cache = self.cache.lock().await;
-        match fragment_cache.entry(url_without_frag.clone()) {
+        let extractor = match FileType::from(path) {
+            FileType::Markdown => extract_markdown_fragments,
+            FileType::Html => extract_html_fragments,
+            FileType::Plaintext => return Ok(true),
+        };
+        match self.cache.lock().await.entry(url_without_frag) {
             Entry::Vacant(entry) => {
                 let content = fs::read_to_string(path).await?;
-                let file_frags = extract_markdown_fragments(&content);
+                let file_frags = extractor(&content);
                 Ok(entry.insert(file_frags).contains(fragment))
             }
             Entry::Occupied(entry) => Ok(entry.get().contains(fragment)),
         }
     }
+
+    fn remove_fragment(mut url: Url) -> String {
+        url.set_fragment(None);
+        url.into()
+    }
```

Comment:
Maybe we should add some tests for that.

Reply:
`remove_fragments` function doesn't do much more than call `Url::set_fragment`, so I don't feel this function would warrant tests.

Reply:
Makes sense. Thanks for the clarification.